src/viam/config/viam-cpp-sdkConfig.cmake.in

```diff
@@ -0,0 +1,37 @@
+@PACKAGE_INIT@
+
+include(CMakeFindDependencyMacro)
+find_dependency(Boost @Boost_VERSION_MAJOR@.@Boost_VERSION_MINOR@ COMPONENTS headers log)
+
+if (@gRPC_FOUND@)
+  find_dependency(gRPC @gRPC_VERSION_MAJOR@.@gRPC_VERSION_MINOR@ CONFIG)
+endif()
+
+# NOTE(acm): It is up to us to decide what components to get if none
+# are specified. Since the common use case should be getting the SDK,
+# interpret no-component as meaning SDK.
+if (NOT viam-cpp-sdk_FIND_COMPONENTS)
+  set(viam-cpp-sdk_FIND_COMPONENTS viamsdk)
+endif()
+
+# TODO(acm): I don't like this manual step where we need to list out
+# the components, it seems like an easy thing to forget.
+set(_viam-cpp-sdk_supported_components viamapi viamsdk)
+
+# TODO(acm): I also don't like that we need to manually set things up
+# so that asking for viamsdk cause you to ask for viamapi too. I'd
+# expect cmake to know that viamsdk dependson viamapi so asking for the
+# one should force you to get the other. But, it doesn't seem to do
+# that, so force it, and force it ahead of viamsdk so that viamsdk's
+# file can resolve viam-cpp-sdk::viamapi
+if (viamsdk IN_LIST viam-cpp-sdk_FIND_COMPONENTS)
+  LIST(PREPEND viam-cpp-sdk_FIND_COMPONENTS viamapi)
+endif()
```

Comment:
This stuff is new. I couldn't figure out how to teach CMake that if you did `find_package(viam-cpp-sdk CONFIG REQUIRED viamsdk)` that it should imply the import of `viamapi` as well. And without it, the `viam-cpp-sdk::viamsdk` component doesn't load because it can't find its `viamapi` dependency. I don't like this code but I don't want to hold things up any further waiting for a silver bullet.